plugins/module_utils/redfish_utils.py

```diff
@@ -3051,3 +3052,60 @@ def get_manager_inventory(self, manager_uri):
 
     def get_multi_manager_inventory(self):
         return self.aggregate_managers(self.get_manager_inventory)
+
+    def set_session_service(self, sessions_config):
+        result = {}
+        response = self.get_request(self.root_uri + self.session_service_uri)
+        if response['ret'] is False:
+            return response
+        current_sessions_config = response['data']
+        payload = {}
+        for property, value in sessions_config.items():
+            value = sessions_config[property]
+            if property not in current_sessions_config:
+                return {'ret': False, 'msg': "Property %s in sessions_config is invalid" % property}
+            if isinstance(value, dict):
+                if isinstance(current_sessions_config[property], dict):
+                    payload[property] = value
+                elif isinstance(current_sessions_config[property], list):
+                    payload[property] = [value]
+                else:
+                    return {'ret': False, 'msg': "Value of property %s in sessions_config is invalid" % property}
```

Comment:
In which conditions would the API ever return an "invalid" value for a property? Do we really need to be suspicious of the API?

Reply:
This would be more likely a configuration issue on the side of the user (where they specified a format for the resource that doesn't conform to its schema definition).

However, it's entirely possible the API has a bug and is not representing the resource per the schema definition.

Reply:
@mraineri I am not very familiar with RedFish API capabilities, but that code seems to be complaining if the type of a property is not a `dict` or a `list`. I find it hard to believe that the API writers would "oops, we sent the wrong response type" - not impossible, I know,  but it is a basic test. I want to believe they test *that* :-)

As of the user side, I reckon whatever the module user does will not or at least should not affect the *type* of a response.

Reply:
> @mraineri I am not very familiar with RedFish API capabilities, but that code seems to be complaining if the type of a property is not a `dict` or a `list`. I find it hard to believe that the API writers would "oops, we sent the wrong response type" - not impossible, I know, but it is a basic test. I want to believe they test _that_ :-)

These types of API bugs slip out more often than you think 😬 

In all seriousness though, I'd like to review this pattern in this module at some point and clean it up a bit (along with some other things that have been on my mind lately). The logic seems a bit odd to me, but this is how the module is currently handling JSON objects within portions of the API. I don't quite understand the "list" check here.

> As of the user side, I reckon whatever the module user does will not or at least should not affect the _type_ of a response.

Exactly; my expectation is if the user provides an invalid type to the API, the API will reject it with an HTTP 400. This is more protective to prevent the module from making the request in the first place. Maybe we shouldn't be as strict here and rely on the API to perform the protection.

Reply:
@russoz I filed an issue to track some of these cleanup items I have on my mind (#5210). Feel free to add to it if you think there are other improvements we can make.